core/metrics-core-api/src/main/java/org/hawkular/metrics/core/api/MetricsService.java

```diff
@@ -84,6 +84,15 @@
 
     Observable<Metric> findMetrics(String tenantId, MetricType type);
 
+    /**
+     *
+     * @param tenantId
+     * @param tags
+     * @param type Optional MetricType, if null is given all matching metrics are returned
```

Comment:
My understanding is as follows.

If no type is set, then #findMetrics returns all metrics which type is a 
user type.

If no type is set, then #findMetricsWithTags returns all metrics having 
the specified tags, regardless of the type.

If I understood well, then I think that if no type is set, 
#findMetricsWithTags should return all metrics having the specified 
tags, and which type is a user type.